File: JavaScript/0682_baseball_game/solution.js
function calculate_points(ops) {
    let result = [];
    for (let char of ops) {
        if (char == 'C') {
            result.pop();
        } else if (char == 'D') {
            result.push(result[result.length - 1] * 2);
        } else if (char == "+") {
            result.push(result[result.length - 1] + result[result.length - 2]);
        } else {
            result.push(Number(char));
        }
    }
    let total = result.reduce((a,b) => a + b);
    return total;
}

const ops = ["5","2","C","D","+"];
console.log(calculate_points(ops));

File: Python/0682_baseball_game/solution.py
def calculate_points(ops):
    result = []
    for char in ops:
        if char == "C":
            result.pop()
        elif char == "D":
            result.append(result[-1] * 2)
        elif char == "+":
            result.append(result[-1] + result[-2])
        else:
            result.append(int(char))
    return sum(result)

ops = ["5","2","C","D","+"]
print(calculate_points(ops))